### ISSUE TITLE
printing html contents

### ISSUE THREAD

**Issue by takoy123** [srehanuddin/Cordova-Plugin-Bluetooth-Printer#33]
I have convert to html to base64 but how to print it using this plugin.
could you please give me step by step to use this plugin.

**Comment by citerio** [comment]
Hey takoy, I wanna do the same thing with this plugin, but I don't know how to convert the html string to base 64 image string, What library did you use to manage it?, after this you have to call PrintImage method.

**Comment by takoy123** [comment]
I used html2canvas library to convert html content to base 64  image string

**Comment by citerio** [comment]
Ok, thanks, and you managed to print html contents using this pluging, right?

**Comment by takoy123** [comment]
Yes, after convert, we will have base 64 image string and then call the
method PrintImage in cordova

**Comment by citerio** [comment]
ok, thanks bro

**Comment by citerio** [comment]
Hey takoy, I already managed to integrate the html2canvas library to the project and the canvas string is being perfectly generated because I checked it out, but when printing, it is a mess, I'm just trying to print a "Hi" text but the printer is printing weird characters. My printer is a Zebra printer (bluetooth handheld printer), Do I have to set a size (width and height) to the HTML document?

**Comment by takoy123** [comment]
I want to see your result. 

Sent from my ASUS

-------- Original Message --------

**Comment by citerio** [comment]
Ok, this is my code:

        let html_string = "<html><head></head><body><p>HI</p></body></html>";
        let iframe=document.createElement('iframe');
        document.body.appendChild(iframe);
        setTimeout(function(){
            var iframedoc=iframe.contentDocument||iframe.contentWindow.document;
            iframedoc.body.innerHTML=html_string;
            html2canvas(iframedoc.body)
            .then(function(canvas) {
                context.imgData = canvas.toDataURL("image/png");                
                alert('testing called: ' + canvas.toDataURL("image/png"));
                //alert('success on html canvas');
            })
            .catch((error) => {alert('error: ' + error)});
        }, 10);

**Comment by citerio** [comment]
![Uploading IMAG6207.jpg…]()

**Comment by citerio** [comment]
![imag6206](https://cloud.githubusercontent.com/assets/9136205/26157409/1e00e532-3ae7-11e7-9e79-56ca132a81e2.jpg)
![imag6207](https://cloud.githubusercontent.com/assets/9136205/26157410/1e219674-3ae7-11e7-84a1-0b1a6b5de08f.jpg)

**Comment by citerio** [comment]
Hey Takoy, did you use this plugin on IONIC 2?, I was searching for another plugin and found out one called "cordova-plugin-datecsprinter-example" but it is not working, because the methods are not being called or are doing nothing.

**Comment by takoy123** [comment]
No. I use it on ionic 1. It is work for me. but I think that it is also working on ionic2. 

Sent from my ASUS

-------- Original Message --------

**Comment by citerio** [comment]
Yeah, but I don't know why the printer is just printing those characters, actually I tested sending directly the base 64 string (var image = 'data:image/png....................................................') to the printImage function and I got the same result, What printer did you use?, I am using a Zebra printer.

**Comment by takoy123** [comment]
it support all type of printer. when I did it, i got the same result like your result. but i try to change some things it work

Sent from my ASUS

-------- Original Message --------

**Comment by takoy123** [comment]
you can copy the base 64 string to test in web site that convert image base 64 string to imase. if it display like original charector it mean that the image base 64 string correctly

Sent from my ASUS

-------- Original Message --------

**Comment by takoy123** [comment]
In your case i thinks your image base 64 string is not correct

Sent from my ASUS

-------- Original Message --------

**Comment by citerio** [comment]
Ok, nice takoy, I was using this snippet http://jsfiddle.net/SEMsq/29/ to test my code and I am getting the base 64 string in the browser developer console with: 
console.log(canvas.toDataURL('image/png'));
then I open that link directly from the console and the browser opens a new tab showing the png image with the Hi text, so I think the base 64 string is perfect, then I copy this string directly in my printImage function and it is just showing those ugly characters. I'm using a iframe element to produce my html document, but I don't know if this is the best way, I just read this post:
http://stackoverflow.com/questions/16116592/html-string-in-html2canvas

**Comment by citerio** [comment]
Did you change the original plugin (Java class)? or Did you create the html document other way?, thanks

**Comment by takoy123** [comment]
function printMyImage() {
  var imagedata;
  html2canvas([document.getElementById('mainDiv')], {
    background: '#fff',
    onrendered: function (canvas) {
      imagedata = canvas.toDataURL('image/png');

      var myImage = new Image();
      myImage.src = imagedata;
      myImage.onload = function () {
        var canvas = document.createElement("canvas");
        canvas.height = 890;
        canvas.width = 380;
        var context = canvas.getContext('2d');
        context.drawImage(myImage, 0, 0);
        var imageBase =
canvas.toDataURL('image/png').replace(/^data:image\/(png|jpg|jpeg);base64,/,
"");
        window.DatecsPrinter.printImage(
          imageBase, //base64
          canvas.width,
          canvas.height,
          0,
          function () {
            $state.go('app.activities');
            $rootScope.txtClear();

            $timeout(function () {

................

..............

..........



there height and

**Comment by citerio** [comment]
Hey Ok, this DatecsPrinter is the other plugin, but I'm gonna use your image/canvas settings on this Cordova-Plugin-BlueTooth-Printer to see if I get good results.

**Comment by wailynnoo** [comment]
@citerio Can you please share your code that ok for image printing with this please?

**Comment by renatosousafilho** [comment]
How can I print the canvas using the height from a content that`s dynamic. For example, a Receipt Order than which according to amout of items in the order increase or decrease the height from a div that will be printed?

**Comment by renatosousafilho** [comment]
i`m trying somethin like that:

```typescript
setTimeout(function(){
                var imagedata;
                html2canvas(iframedoc.body)
                .then(function(canvas) {
                  imagedata = canvas.toDataURL('image/png');
                  var myImage = new Image();
                  myImage.src = imagedata;
                  myImage.onload = function () {
                    var canvas = document.createElement("canvas");
                    canvas.width = 365;
                    var body = iframedoc.body;
                    var html = iframedoc.documentElement;
                    var height = Math.max( body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight );
                    canvas.height = height + 260; // works increase the size with static value.
                    var context = canvas.getContext('2d');
                    context.drawImage(myImage, 0, 0);
                    var imageBase =
                    canvas.toDataURL('image/png').replace(/^data:image\/(png|jpg|jpeg);base64,/,"");

                    $self.printImageProvider.printImage(imageBase, canvas.width, canvas.height, 0).then((result)=>{
                      $self.printImageProvider.disconnect();
                    });
                  };
                })
                .catch((error) => {alert('erro no html2canvas: ' + error)});
            }, 1000);
```

**Comment by sarancruzer** [comment]
@citerio Can you please share your code that ok for image printing with this please?
